### DISCUSSION TITLE
Is it possible to set a thread-like barrier between builds

### DISCUSSION THREAD

**Discussion by charlie45000** [buildbot/buildbot#7712] [Q&A] [answered]
Hello,

I am curently working on the building configuration and I can't see how to do the following thing :

When I build a release, I want to build and test on three builders:
- one on linux in debug mode
- one on linux in release mode
- one on windows in release mode too.

(the two release builds package too)

once all (on only if all) the builds have passed, I would like to upload on a store the package of the two release builds 
I have looked for possibilities to do this with triggerable schedulers, but this can't be done easily since there are multiple workers for each builder and that I can't guarantee that the second triggered build will run on the same worker ta keep the built packages.

One way to do it would be to have some kind of barrier step : 
the builder waits for others to finish or wait at the barrier too and once all the builders are stopped or wating, if no builder failed, the upload step can be run. (to speed up the process and avoid occupying workers unnecessarily, It could be possible to stop waiting builders as soon as an other builder have failed.

Is there a built-in way to do it? how could it be possible to do it if is isn't built-in?

Thank-you for your help!

**Comment by vibbo** [comment, marked as answer]
Hi.

One way to do it is to upload the artifacts of each build to artifactory/shared_disk/etc. and if each build passed you can "release" from that location(s). Or delete if any of them failed.

**Reply by charlie45000** [reply]
Hi,

I did something like that but using triggered builds from a """master builder""" local to the buildbot master and file transfer from the workers to the master, but it's less practical since if any additional file is needed for the release in the future it will necessitate maintenance...

But I suppose there is no choice (other than creating a barrier step from scratch).

Thank you for your reply!
